explainer-datamodel.md

```diff
@@ -0,0 +1,413 @@
+be supported because it has the effect of changing the object's computed display value 
+from inline to block (taking it out-of-flow). `position: absolute` also takes
+the object out of the normal flow. Likewise, the `width` and `height` properties would
+be ignored because they do not apply to inline-level elements. Attempting to set these 
+properties would have no effect. On the other hand, `padding` would be supported and 
+applied to `FormattedTextRun` objects in accordance with the rules of CSS inline layout,
+e.g., the inline-direction `padding` values are taken into account while the 
+block-direction values are not.
+
+The `FormattedText` object is an independent containing block for text. It is essentially a 
+`display: flow-root` object that establishes an inline formatting context for its children.
+The `FormattedText` object has no positional relationship with anything other than its
+children, and so values like `top`, `right`, `bottom`, and `left` do not apply (i.e., the 
+object acts as its own initial containing block). Positioning the object for the purpose of
+rendering in some context must be done separately. In order to render or extract useful metrics
+from the `FormattedText` object, final dimensions for its size must be supplied (see 
+[Rendering explainer](explainer-rendering.md)) which are used to resolve relative values in
+supported properties like `width` and `height`.
+
+In some cases, we imagine it will be useful to allow the `FormattedText` object to assume
+alternate layout container types where those alternate types provide unique text layout
+capabilities. For example, we expect to support `FormattedText` objects with an inner 
+display type of `ruby` in order to become a Ruby container and enable the use of Ruby
+annotated layout. Other container types are not currently planned to initially support 
+but are good long-term candidates (e.g., multi-column containers created via the `columns`
+shorthand property), while still others are less-likely to be supported (e.g., flex and 
+grid container types, which are less useful for formatted text scenarios).
+
+There are various CSS properties that provide helpful graphical emphasis to text that are 
+also supported. These are for convenience in supporting common text formatting scenarios
+that would otherwise require detailed introspection of the object model's related metrics
+in order to correctly layout and render as desired with respect to the text. Because these
+features are already available in CSS layout engines and significantly ease author burden, 
+many of these CSS properties will be supported. Some supported examples include: 
+`text-decoration`, `text-shadow`, `box-shadow`, even `border`, `outline`, and limited `background` support 
+(where the metrics and composition processing do not require external dependencies, such as 
+image resources typically loaded by `url()` functions).
+
+#### Text clipping and future extensions 
```

Comment:
Can we call this Future Extensions, without calling out text clipping. Supporting clip-path for example does not work in the implementation. I am not sure how much work is involved in making it work.

Reply:
Sounds good.

Reply:
See https://github.com/WICG/canvas-formatted-text/commit/1ac4886ac55b5d60cdb4f6391d218ae28d5e4e73